### ISSUE TITLE
UnusedVariable false positive on private record class

### ISSUE THREAD

**Issue by olsavmic** [google/error-prone#3495]
Following code reports:

`
[WARNING] /Users/michael/Dev/Work/error-prone-testing/src/main/java/org/example/Main.java:[16,13] [UnusedVariable] The parameter 'value' is never read.
    (see https://errorprone.info/bugpattern/UnusedVariable)
`

```java
class Main
{

    public static void main(String[] args)
    {
        var data = new Data(10);
        data.value();
    }

    private record Data(
        int value
    )
    {

    }

}
```

Changing the record visibility to package-private solves the issue but is not optimal.

com.google.errorprone.error_prone_core: 2.16
JDK 17, source & target 17

**Comment by tbroyer** [comment]
AFAICT this is a duplicate of #2713

**Comment by olsavmic** [comment]
Thanks, I was somehow not able to find it, closing in favor of the existing one :)
